src/app/shared/api/food/food.service.ts

```diff
@@ -21,11 +21,11 @@ export class FoodApiService {
   }
 
   public addFoodItem(food: Food): Promise<Food> {
-    return (this.foodItems.push(food) as any) as Promise<Food>;
+    return this.foodItems.push(food) as any as Promise<Food>;
```

Comment:
Is the intermediary `as any` required?

Reply:
Seems it's there to silence a type conversion warning.
Turns out `.push()` doesn't return a `Promise<Food>`.